File: api/Medico.Api/DB/Models/PatientDemographic.cs
using System;
using System.Collections.Generic;
using Medico.Api.DB.Enums;

namespace Medico.Api.DB.Models
{
    public class PatientDemographic : CompanyRelatedEntity
    {
        public BaseVitalSigns BaseVitalSigns { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string MiddleName { get; set; }

        public Gender Gender { get; set; }

        public DateTime DateOfBirth { get; set; }

        public MaritalStatus MaritalStatus { get; set; }

        public string Ssn { get; set; }

        public string PrimaryAddress { get; set; }

        public string SecondaryAddress { get; set; }

        public string City { get; set; }

        public string PrimaryPhone { get; set; }

        public string SecondaryPhone { get; set; }

        public string Email { get; set; }

        public string Zip { get; set; }

        public State State { get; set; }

        public Guid? PatientInsuranceId { get; set; }

        public PatientInsurance PatientInsurance { get; set; }

        public List<Appointment> Appointments { get; set; }

        public List<Admission> Admissions { get; set; }

        public List<MedicalHistory> MedicalHistory { get; set; }

        public List<SurgicalHistory> SurgicalHistory { get; set; }

        public List<FamilyHistory> FamilyHistory { get; set; }

        public List<EducationHistory> EducationHistory { get; set; }

        public List<OccupationalHistory> OccupationalHistory { get; set; }

        public List<Allergy> Allergies { get; set; }

        public List<MedicationHistory> MedicationHistory { get; set; }

        public List<MedicalRecord> MedicalRecords { get; set; }

        public List<TobaccoHistory> TobaccoHistory { get; set; }

        public List<AlcoholHistory> AlcoholHistory { get; set; }

        public List<VitalSigns> VitalSigns { get; set; }

        public List<DrugHistory> DrugHistory { get; set; }
    }
}

File: client/src/dataModels/patientDemographic.ts

import { BaseEntityModel } from "./baseEntityModel";
import { DateConverter } from "../helpers/dateConverter";

export class PatientDemographic extends BaseEntityModel {
    FirstName: string;
    MiddleName: string;
    LastName: string;
    PrimaryAddress: string;
    SecondaryAddress: string;
    City: string;
    State: number;
    Zip: string;
    PrimaryPhone: string;
    SecondaryPhone: string;
    Gender: number;
    MaritalStatus: number;
    Ssn: string;
    Email: string;
    DateOfBirth: any;
    PatientInsuranceId: string;
    CompanyId: string;

    constructor(id: string = "", isDelete: boolean = false) {
        super(id, isDelete);

        this.FirstName = "";
        this.MiddleName = "";
        this.LastName = "";
        this.PrimaryAddress = "";
        this.SecondaryAddress = "";
        this.City = "";
        this.State = null;
        this.Zip = "";
        this.PrimaryPhone = "";
        this.SecondaryPhone = "";
        this.Gender = null;
        this.Ssn = "";
        this.DateOfBirth = null;
        this.MaritalStatus = null;
        this.PatientInsuranceId = "";
        this.Email = "";
        this.CompanyId = "";
    }

    convertToEntityModel() {
        this.DateOfBirth = DateConverter
            .jsLocalDateToSqlServerUtc(this.DateOfBirth);
    }

    createFromEntityModel(entity: any) {
        const newPatient = new PatientDemographic(entity.Id, !!entity.IsDelete);

        newPatient.FirstName = entity.FirstName;
        newPatient.MiddleName = entity.MiddleName;
        newPatient.LastName = entity.LastName;
        newPatient.PrimaryAddress = entity.PrimaryAddress;
        newPatient.SecondaryAddress = entity.SecondaryAddress;
        newPatient.City = entity.City;
        newPatient.State = entity.State;
        newPatient.Zip = entity.Zip;
        newPatient.PrimaryPhone = entity.PrimaryPhone;
        newPatient.SecondaryPhone = entity.SecondaryPhone;
        newPatient.Gender = entity.Gender;
        newPatient.Ssn = entity.Ssn;
        newPatient.PatientInsuranceId = entity.PatientInsuranceId;
        newPatient.MaritalStatus = entity.MaritalStatus;
        newPatient.CompanyId = entity.CompanyId;
        newPatient.Email = entity.Email;

        newPatient.DateOfBirth = DateConverter
            .sqlServerUtcDateToLocalJsDate(entity.DateOfBirth);

        return newPatient;
    }
}